File: frontend/icon/src/types/JobListItem.ts
import { JobRunStatus, JobStatus } from "./enums";

/**
 * A job list element. Represents the data shown in the job list.
 */
export interface JobListItem {
  id: number;
  created: string;
  status: JobStatus;
  experiment_id: string;
  num_scan_parameters: number;
  /** ID of the job's latest run, used to route `job_run.update` events. */
  run_id: number | null;
  /** Status of the latest run, or null if the job has not run yet. */
  run_status: JobRunStatus | null;
}

File: src/icon/server/data_access/job_list_item.py
from typing import TypedDict


class JobListItemDict(TypedDict):
    """A job list item. Minimal, non ORM representation of a job, suitable for fast serialization."""

    id: int
    """Database identifier of the job."""
    created: str
    """Creation timestamp in ISO format."""
    status: str
    """Job status, e.g. "submitted", "processing" or "processed"."""
    experiment_id: str
    """Identifier of the experiment this job runs."""
    num_scan_parameters: int
    """Number of scanned parameters, rendered as the scan dimensionality."""
    run_id: int | None
    """Identifier of the job's latest run, or None if it has not run yet."""
    run_status: str | None
    """Status of the latest run, or None if the job has not run yet."""
